.travis.yml

```diff
@@ -1,14 +1,37 @@
-language: go
-go:  
-   - "1.8.x"
-   - "1.9.x"
-   - "1.10.x"
-   - "1.11.x"
-before_script: 
-    - export GOPATH="$HOME/gopath"
-    - go get -d -v
-    - make prepare
-    - "export LD_LIBRARY_PATH=$PWD/libgit2/install/lib"
-    - "export PKG_CONFIG_PATH=$PWD/libgit2/install/lib/pkgconfig"
-script: 
-   - make test
+services:
+- docker
+
+matrix:
+  include:
+  - os: linux
+    dist: bionic
+    env: TARGET_OS_ARCH=linux/amd64
+  - os: linux
+    dist: bionic
+    env: TARGET_OS_ARCH=darwin/amd64
+  - os: linux
+    dist: bionic
+    env: TARGET_OS_ARCH=windows/amd64
+  - os: linux
+    dist: bionic
+    env: TARGET_OS_ARCH=windows/386
+
+script:
+- docker run -it -v $PWD:$PWD -w $PWD navigaid/osxcross ./static_build.sh ${TARGET_OS_ARCH}
+
+before_deploy:
+- tar czf gitql-${TRAVIS_TAG:-${TRAVIS_COMMIT}}-${TARGET_OS_ARCH////-}.tar.gz gitql*
+- shasum -a 256 gitql-${TRAVIS_TAG:-${TRAVIS_COMMIT}}-${TARGET_OS_ARCH////-}.tar.gz > gitql-${TRAVIS_TAG:-${TRAVIS_COMMIT}}-${TARGET_OS_ARCH////-}.tar.gz.checksum.txt
+
+deploy:
+  api_key:
+    secure: YdM/HjLuNXXgJu4M1Vc66yORh4BeOP36D+jLy3sbE3h4c00QyCV8NEmmJOCjH/c+YZsJsESmvdSxhKxFcaRHvWslUyV9LiG6H0xRXJAsd295aBC7KVPOJEyoNcIKDT02Axh9y5DRIH3Jk4jI6lxVkU5m4ewR8LWwrkXQS1lGTDLrtrGB1ymmbmjKmxbM/ZkyVS98cPB/QKA+P6GxO2KkQzfJ8ExfEa/SOWUxxytSa14Brypd3RHIVsCmQUcgZEEsChUdQsddPLDe8tgKFMnOvWF6U+B9yoltMU99g61sdiLBMS4MSRViivSmPZCUEmzukasCwPP0jzZ1KfZGP2YLy+WBDBGCl0Qs6lz2/ItSSqW24K+60duKox9QiGqMObXBvorMo/07n1Liao6SP1RYM6eghmObviplxth8LEruVR1AHRYCrbPTU0z6RV1r+5MLTzilIUB5ZyU6OCBr8siLjvomdvss0+dYXUQa2bBLcWwb3dnrTEUy8Bjv9WHSOKM2QwmTF52RsLV5LHIRBw2rQVMa+VVvDBzpzCqmtCgUYxeOw4/rb8T0sPeRQFX2vMi58kdwBdXOjxbBF/x1LIIVGcM8WVY58x3XnVXgxU3irFqUcT5tSKk7f0LIm+CVKijmEoBMs8jwmShJZyHkGp7MwHPB3lrphN/gn3bHeJDepZ4=
```

Comment:
Is this key something that we must need to commit? It seems some specific for you travis account.

Reply:
It's specific to my travis account, you need to replace it with your own if you want automated github release. Here is a detailed instruction on how to setup your own secret api_key: https://stackoverflow.com/questions/9338428/using-secret-api-keys-on-travis-ci

Reply:
Can we have the API Key stored as an environment variable instead in the settings?

[Travis ref](https://docs.travis-ci.com/user/environment-variables/#defining-variables-in-repository-settings)

Reply:
@jsixface I think it's common practice to commit the encrypted secret along with .travis.yml, you can find plenty of examples here: https://github.com/search?q=filename%3A.travis.yml+api_key%3A+secure%3A&type=Code

Of course it doesn't make any sense to commit my encrypted api_key to this repo. I think you can post your encrypted key here and I will replace mine with yours @cloudson in a follow up commit.